rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,100 @@
+### Users should be able to upload their own gene sets and see them in the right side bar with the option to apply them within current cellxgene functionalities, if the same gene ontology was used when generating the current dataset and the geneset uploaded by the user (ie the gene names match) 
+- [US2] A user should be able to upload a geneset via a csv file (see data formats below for more detail) for a dataset.
+- [US2.5] A user should be able to upload a geneset they previously downloaded from cellxgene
+- [UC3] When a user uploads a geneset, that list of genes should appear in the right side bar 
+- [US4] If the user was logged in when they uploaded a geneset that genesit should persist if they logout and back in or reload the page
+- [US5] If the gene names match those used in the dataset the user should be able to use general cellxgene functionality (color by genes, calculate differential expression levels, etc.) on those genes
+- [US6] Genesets uploaded for one dataset should be available in other datasets?
+
+### If there is a standard format for differential expression output, we should follow that. If not, the format should be as straightforward as possible while serving these above cases.
+
+### [Optional] Nonfunctional requirements
+- Users should be able to export precomputed differential expression results in real time
+- Users should be able to upload gene sets and have them immediately appear in the right side bar
+
+## Detailed Design | Architecture | Implementation
+
+
+### [Optional] Data model
+
+Fully supporting persistence of differential expression results is out of the scope of this RFC. But the data model described below has been designed to allow for differential expression persistence in the future.
+###Relational Database
+To support gene sets five tables will be added to the cellxgene relational database
+- Gene
+   - UUID (generated)
+   - GeneName
+   - OntologyId?
+- GeneGeneSetLink
+   - UUID
+   - Gene UUID
+   - GeneSet UUID
+   - Comments (optional)
+- GeneSet
+   - UUID (generated)
+   - GeneSet Name (required)
+   - Gene Count (generated)
+- GeneSetUserLink
+   - User UUID
+   - GeneSet UUID
+   - Comments (optional)
+   - Dataset UUID?
+-GeneSetDatasetLink
+   - UUID (generated)
+   - GeneSet UUID
+   - Dataset UUID
+
+Note - GeneSets will only be linked to datasets when they were uploaded with that dataset
+![Cellxgene Data Schema](imgs/Cellxgene_rds_schema.png)
+**Figure 1** Cellxgene data schema, tables to be added (as described above) are in red
+#### Imported CSV
+GeneSet CSV
+- A file containing a comma separated list of the gene set name, genes, pvalue, logfold value, 
+identifers for the cellsets that were compared to produce those values and details about why they were included. 
+Each gene should be on a new line. Only gene set (name) and gene are required fields. The name of the file will be based on the 
```

Comment:
What does the CSV look like without the optional fields?

This:
SET1, a23,,,,,
SET1, b46,,,,,

Or this:
SET1, a23
SET1, b46

Reply:
The top, updated the example for clarity